src/nbl/video/ILogicalDevice.cpp

```diff
@@ -7,27 +7,25 @@
 using namespace nbl;
 using namespace nbl::video;
 
-static void debloatShaders(const asset::ISPIRVDebloater& debloater, std::span<const asset::IPipelineBase::SShaderSpecInfo> shaderSpecs, core::smart_refctd_ptr<asset::IShader>* outShaders, asset::IPipelineBase::SShaderSpecInfo* outShaderSpecInfos)
+static void debloatShaders(const asset::ISPIRVDebloater& debloater, std::span<const asset::IPipelineBase::SShaderSpecInfo> shaderSpecs, core::vector<core::smart_refctd_ptr<const asset::IShader>>& outShaders, asset::IPipelineBase::SShaderSpecInfo* outShaderSpecInfos)
 {
     for (const auto& shaderSpec: shaderSpecs)
     {
-        const auto entryPoint = asset::ISPIRVDebloater::EntryPoint{
+        const core::set<asset::ISPIRVDebloater::EntryPoint> entryPoints = {
+          {
             .name = shaderSpec.entryPoint,
-            .shaderStage = shaderSpec.stage,
+            .stage = shaderSpec.stage,
+          },
         };
-        if (shaderSpec.shader)
-        {
-          const auto* shader = shaderSpec.shader;
-          *outShaders = core::make_smart_refctd_ptr<asset::IShader>(debloater.debloat(shader->getCode(), std::span(&entryPoint, 1)), shader->getContentType(), std::string(shader->getFilepathHint()));
-        } else
+        const auto* shader = shaderSpec.shader;
+        auto debloatedShaderSpec = shaderSpec;
+        if (shader != nullptr)
         {
-          *outShaders = nullptr;
+            outShaders.push_back(debloater.debloat(shader, entryPoints));
+            debloatedShaderSpec.shader = outShaders.back().get();
```

Comment:
its somewhat dangerous to write code like this, I get that the reserve is supposed to prevent reallocation, but you better assert that `data()` before the push_back is exactly the same as `data()` after